util/src/main/java/org/datacommons/util/CoordinatesResolver.java

```diff
@@ -0,0 +1,112 @@
+package org.datacommons.util;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.Values;
+import org.datacommons.proto.Mcf.ValueType;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest.Coordinate;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse;
+
+/** Resolves nodes with lat-lngs by calling the DC coordinates resolution API. */
+public class CoordinatesResolver {
+  private final AtomicBoolean resolved = new AtomicBoolean(false);
+
+  private final Set<Coordinate> resolveCoordinates = ConcurrentHashMap.newKeySet();
+
+  private final ConcurrentHashMap<Coordinate, Set<String>> resolvedCoordinates =
+      new ConcurrentHashMap<>();
+
+  private final ReconClient client;
+
+  public CoordinatesResolver(ReconClient client) {
+    this.client = client;
+  }
+
+  public void submitNode(PropertyValues node) {
+    if (resolved.get()) {
+      throw new IllegalStateException("submitNode called after remote resolution.");
+    }
+    getCoordinate(node).ifPresent(resolveCoordinates::add);
+  }
+
+  // TODO: Pick the ID based on a preferred list.
+  public String resolveNode(PropertyValues node) {
+    return getCoordinate(node)
+        .filter(resolvedCoordinates::containsKey)
+        .flatMap(coordinate -> resolvedCoordinates.get(coordinate).stream().findFirst())
+        .orElse("");
+  }
+
+  // TODO: Support chunking in batches of max size 500.
+  public void remoteResolve() throws IOException, InterruptedException {
+    if (resolved.get()) {
+      throw new IllegalStateException("remoteResolve called after remote resolution.");
+    }
+    resolved.set(true);
+
+    if (resolveCoordinates.isEmpty()) {
+      return;
+    }
+
+    ResolveCoordinatesRequest request =
+        ResolveCoordinatesRequest.newBuilder().addAllCoordinates(resolveCoordinates).build();
+    ResolveCoordinatesResponse response = client.resolveCoordinates(request);
```

Comment:
Where are we making the call?

Also, should we limit the number of things added?

Reply:
I was planning on adding chunking in the next PR but have done it in this PR itself. PTAL.